xarray/namedarray/core.py

```diff
@@ -844,5 +853,141 @@ def _to_dense(self) -> NamedArray[Any, _DType_co]:
         else:
             raise TypeError("self.data is not a sparse array")
 
+    def permute_dims(
+        self,
+        *dims: Hashable | ellipsis,
+        missing_dims: ErrorOptionsWithWarn = "raise",
+    ) -> NamedArray[Any, _DType_co]:
+        """Return a new object with transposed dimensions.
+
+        Parameters
+        ----------
+        *dims : Hashable, optional
+            By default, reverse the dimensions. Otherwise, reorder the
+            dimensions to this order.
+        missing_dims : {"raise", "warn", "ignore"}, default: "raise"
+            What to do if dimensions that should be selected from are not present in the
+            NamedArray:
+            - "raise": raise an exception
+            - "warn": raise a warning, and ignore the missing dimensions
+            - "ignore": ignore the missing dimensions
+
+        Returns
+        -------
+        transposed : NamedArray
+            The returned object has transposed data and dimensions with the
+            same attributes as the original.
+
+        Notes
+        -----
+        This operation returns a view of this variable's data. It is
+        lazy for dask-backed Variables but not for numpy-backed Variables.
+
+        See Also
+        --------
+        numpy.transpose
+        """
+        from xarray.core.indexing import as_indexable
+
+        if not dims:
+            dims = self.dims[::-1]
+        else:
+            dims = tuple(infix_dims(dims, self.dims, missing_dims))
+
+        if len(dims) < 2 or dims == self.dims:
+            # no need to transpose if only one dimension
+            # or dims are in same order
+            return self.copy(deep=False)
+
+        axes = self.get_axis_num(dims)
+        data = as_indexable(self._data).transpose(axes)  # type: ignore
+        return self._replace(dims=dims, data=data)
+
+    @property
+    def T(self) -> NamedArray[Any, _DType_co]:
+        return self.permute_dims()
+
+    def _check_dims(self, dims: Mapping[Any, _Dim] | _Dim):
+        if isinstance(dims, dict):
+            dims_keys = dims
+        else:
+            dims_keys = dims if isinstance(dims, str) else list(dims)
+        if missing_dims := set(self.dims) - set(dims_keys):
+            raise ValueError(
+                f"new dimensions {dims!r} must be a superset of "
+                f"existing dimensions {self.dims!r}. missing dims: {missing_dims}"
+            )
+
+    def _get_expanded_dims(self, dims: Mapping[Any, _Dim] | _Dim) -> _DimsLike:
+        self_dims = set(self.dims)
+        return tuple(dim for dim in dims if dim not in self_dims) + self.dims
+
+    def broadcast_to(self, dims: Mapping[Any, _Dim]) -> NamedArray[Any, _DType_co]:
```

Comment:
```suggestion
    def broadcast_to(self, dims: Mapping[Any, _Dim], **dims) -> NamedArray[Any, _DType_co]:
```

Would be nice to allow `na.broadcast_to(time=4, lat=4)` following most of our other functions

Reply:
addressed